src/safari-app-extension/AlpheiosReadingTools/AppDelegate.swift

```diff
@@ -82,41 +82,46 @@ class AppDelegate: NSObject, NSApplicationDelegate {
         return container
     }()
     
-    // MARK: - Core Data Saving and Undo support
-    
-    @IBAction func saveAction(_ sender: AnyObject?) {
-        #if DEBUG
-        os_log("Save action callback is executiong", log: OSLog.sAlpheios, type: .info)
-        #endif
-        
-        // Performs the save action for the application, which is to send the save: message to the application's managed object context. Any encountered errors are presented to the user.
-        let context = persistentContainer.viewContext
+    func fetchAuthInfo() {
+        // Obtain a managed context
+        let managedContext = self.persistentContainer.viewContext
+        // Create a fetch request
+        let fetchRequest = NSFetchRequest<NSManagedObject>(entityName: "AuthUser")
         
-        if !context.commitEditing() {
-            NSLog("\(NSStringFromClass(type(of: self))) unable to commit editing before saving")
-        }
-        if context.hasChanges {
+        do {
+            // Check how many records are in the store
+            let recordsInStore = try managedContext.count(for: fetchRequest)
             #if DEBUG
-            os_log("Store context has changes", log: OSLog.sAlpheios, type: .info)
+            os_log("Persistent store has %d user records", log: OSLog.sAlpheios, type: .info, recordsInStore)
             #endif
-            do {
-                try context.save()
-                #if DEBUG
-                os_log("Store context has been changed", log: OSLog.sAlpheios, type: .info)
-                #endif
-            } catch {
-                // Customize this code block to include application-specific recovery steps.
-                let nserror = error as NSError
-                os_log("Cannot save tore context: %@", log: OSLog.sAlpheios, type: .error, nserror)
-                NSApplication.shared.presentError(nserror)
+            
+            // Returns an array of managed objects meeting the criteria specified by the fetch request
+            let users = try managedContext.fetch(fetchRequest)
+            for user in users {
```

Comment:
would there ever be a valid scenario where we have multiple users?

Reply:
We should never have multiple users. However, Core Data allows to store multiple objects. So I was thinking if anything will go wrong due to some unexpected app behavior (i.e. an app crash or something like that) we might (although very unlikely) end up with multiple records in the database. In that case it's probably good to have code that will not go bananas seeing multiple user records. What do you think?

Reply:
agree